src/librustc_lint/unused.rs

```diff
@@ -143,7 +143,7 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for UnusedResults {
                 return true;
             }
 
-            let plural_suffix = if plural { "s" } else { "" };
```

Comment:
Instead of changing the macro `pluralize!()` to take a boolean I would change this to `pluralize!(if plural { 2 } else { 1 })`. The local ugliness is outweighed by the alternative ugliness throughout the codebase.

Reply:
I believe we can do better and actually [carry around the length](https://github.com/rust-lang/rust/pull/64342#pullrequestreview-286083484) value here, instead of a boolean, with a modicum of refactoring.